### PR TITLE
CORTX-28871: Modify admin user retry logic into rgw_lock block

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

# Problem Statement
- rgw_setup config phase failed since rgw_user_create method does not return anything if lock is acquired

# Design
-  Move retry logic into rgw_lock=True block

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by shriya-deshmukh — APPROVED** [review]

File: src/setup/rgw.py [hunk 1]
```diff
@@ -538,13 +524,30 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf, data_pod_hostname: s
         if rgw_lock is True:
             Log.info('Creating admin user.')
             # Before creating user check if user is already created.
-            status = Rgw._create_rgw_user(conf)
-            if status == 0:
+            user_status = Rgw._create_rgw_user(conf)
+
+            if user_status != 0:
+                current_data_node = Rgw._get_current_data_node()
```
Comment (cdeshmukh) [inline src/setup/rgw.py:530]: its already available as data_pod_hostname, can reuse

File: src/setup/rgw.py [hunk 2]
```diff
@@ -538,13 +524,30 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf, data_pod_hostname: s
...
+                machine_ids = Rgw._get_cortx_conf(conf, 'cluster>storage_set[0]>nodes')
+                data_pod_hostnames = [Rgw._get_cortx_conf(conf, \
+                    f'node>{machine_id}>hostname') for machine_id in machine_ids if \
+                    Rgw._get_cortx_conf(conf, f'node>{machine_id}>type') == 'data_node']
+                data_pod_hostnames.remove(current_data_node)
+                for data_pod_hostname in data_pod_hostnames:
+                    status = Rgw._update_hax_endpoint_and_create_admin(conf, \
```
Comment (cdeshmukh) [inline src/setup/rgw.py:537]: why this recursive call? Lock is there, we can just loop around the required logic !!

File: src/setup/rgw.py [hunk 3]
```diff
@@ -137,22 +137,8 @@ def config(conf: MappedConf):
         # config phase since data pod starts before server pod.
         # Try HAX endpoint from data pod of same node first & if it doesnt work,
         # from other data pods in cluster
-        current_data_node = socket.gethostname().replace('server', 'data')
```
Comment (cdeshmukh) [inline src/setup/rgw.py:140]: why this move is needed exactly to line 530? Please add explanation here (not in code)

**Review by cdeshmukh — CHANGES_REQUESTED** [review]

File: src/setup/rgw.py (hunk 1 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py:530]: No, data_pod_hostname will change on recursive calls

**Review by sachitanands — APPROVED** [review]
apart from the changes suggested , rest of the code looks good.

File: src/setup/rgw.py (hunk 3 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py:140]: It is used both in 140 & 530. Hence created a new function..
when recursively called data_pod_hostname changes to other pod's hostname

File: src/setup/rgw.py (hunk 2 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py:537]: Can be done in 2 ways.
1. recursive call. (Reads other data pods information only if current data pod fails.)
2. Read all data_pods and iterate till it returns success.

File: src/setup/rgw.py (hunk 3 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py:140]: Code is optimised by creating two separate functions

File: src/setup/rgw.py (hunk 2 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py]: Done.

File: src/setup/rgw.py [hunk 4]
```diff
@@ -458,8 +443,8 @@ def _generate_ssl_cert(conf: MappedConf):
                 raise SetupError(errno.EINVAL, f'Failed to generate self signed ssl certificate: {e}')
 
     @staticmethod
-    def _update_hax_endpoint_and_create_admin(conf: MappedConf, data_pod_hostname: str):
-        """Update motr_ha(hax) endpoint values to rgw config file."""
+    def _update_hax_endpoint(conf: MappedConf, data_pod_hostname: str):
```
Comment (cdeshmukh) [inline src/setup/rgw.py:446]: data_pod_hostname is not being validated at all, like it should not be a null value or empty sting, in which case it should return an non-zero error code and consuming function should skip that data_pod_hostname

Comment (cdeshmukh) [reply src/setup/rgw.py:446]: @selvakumaar5496 please address above comment for error handling separately. rest of the code looks fine, merging now, so we can trigger the build, to use for other imp tasks

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**